### ISSUE TITLE
Query document by date

### ISSUE THREAD

**Issue by alanpachuau** [louischatriot/nedb#383]
I have my persons document stored something like this

`{"name":"Alan` Pachuau","birthday":{"$$date":823631400000},"_id":"NmkakE7aHmJ19Xv5"}
`{"name":"John` Pachuau","birthday":{"$$date":623631400000},"_id":"NmkakE7aHmJ19Xv5"}

I want to query the documents using the _birthday_ field. I am trying to find person having birthday today. I couldn't find how to do this in the documentation. Any help would be appreciated.

Thank you

**Comment by frenchbread** [comment]
Try to store birthday's timestamp as a number. E.g.:
```json
{ "name" : "Alan Pachuau", "birthday" : 823631400000 }, "_id" : "NmkakE7aHmJ19Xv5"}
```
This way, you can also filter by date range, like [here](http://stackoverflow.com/a/2943685).

**Comment by bdfoster** [comment]
Do not store a date as a number. Store it as a Date object:

From the documentation:
var doc = { hello: 'world' , n: 5 , today: new Date() , nedbIsAwesome: true
, notthere: null , notToBeSaved: undefined // Will not be saved , fruits: [
'apple', 'orange', 'pear' ] , infos: { name: 'nedb' } };

That Date object works just like a JavaScript Date object (because it is).

There are *4 ways* of initiating a date:

new Date()
new Date(milliseconds)
new Date(dateString)
new Date(year, month, day, hours, minutes, seconds, milliseconds)

http://www.w3schools.com/js/js_dates.asp

**Comment by frenchbread** [comment]
@bdfoster Will the filtering work with `Date()` object approach just like in MondoDB (using `$lte and $gte` )?

**Comment by bdfoster** [comment]
I'd imagine it would, but I'm not in a position where I can test easily. He
doesn't need to use a filter though.

**Comment by frenchbread** [comment]
@bdfoster Yep, I was asking for myself since didn't want to create separate issue.

**Comment by bdfoster** [comment]
Your referenced link says the same thing, to use ISODate to store a date.
ISODate is a wrapper over the Date object.

**Comment by frenchbread** [comment]
@bdfoster I needed to use filtering to fetch documents in specific range of dates. `Date()` obj didn't work "out of the box". And for now I've decided to go with unix timestap. Even though I'm gonna try it out with ISODate again.

**Comment by bdfoster** [comment]
You may not doing it correctly, but remember you can get epoch out of a
Date object. If you're struggling, open another issue and post what you've
done.

**Comment by frenchbread** [comment]
Filtering by ISODate works well. The issue was that I've been storing `moment()` object in the db instead of ISODate. `moment.toDate() // converts to ISO Date` did the thing.

**Comment by alanpachuau** [comment]
Thank you all for your response. @frenchbread so are you saying `$lte`, `$gte` works with the ISO Date?

**Comment by frenchbread** [comment]
@alanpachuau Yep.

**Comment by rayfoss** [comment]
Runkit confirming creating and finding date with $lt, $gt, etc filters works without issue:
https://runkit.com/hesygolu/5ae24c7e67538600120112b0

Should probably close this.
